kclvm/tools/src/LSP/src/request.rs

```diff
@@ -145,7 +173,16 @@ pub(crate) fn handle_reference(
 ) -> anyhow::Result<Option<Vec<Location>>> {
     let file = file_path_from_url(&params.text_document_position.text_document.uri)?;
     let path = from_lsp::abs_path(&params.text_document_position.text_document.uri)?;
-    let db = snapshot.get_db(&path.clone().into())?;
+    let db = match snapshot.get_db(&path.clone().into())? {
+        Some(db) => db,
+        None => {
+            log_message(
```

Comment:
Too many duplicate codes, please refer to the comments above about the VFS filter.